File: src/main/java/macro/library/external/Goodreads.java
package macro.library.external;

import macro.library.author.Author;
import macro.library.book.Book;
import macro.library.book.Isbn;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Macro303 on 2019-Oct-30
 */
public abstract class Goodreads {
	@NotNull
	private static final Logger LOGGER = LogManager.getLogger(OpenLibrary.class);
	@NotNull
	private static final String URL = "http://openlibrary.org/api/books";

	@Nullable
	public static Book searchBook(@NotNull Isbn isbn) {
		return null;
	}

	@NotNull
	public static List<Author> searchAuthors(@NotNull Isbn isbn) {
		var authorList = new ArrayList<Author>();
		return authorList;
	}
}

File: src/main/kotlin/macro/library/external/Goodreads.kt
package macro.library.external

import macro.library.Util
import macro.library.book.Book
import macro.library.book.Isbn
import macro.library.config.Config.Companion.CONFIG
import org.apache.logging.log4j.LogManager
import org.json.XML

/**
 * Created by Macro303 on 2019-Nov-13
 */
object Goodreads {
	private val LOGGER = LogManager.getLogger(Goodreads::class.java)
	private val URL = "https://www.goodreads.com/book"

	fun searchBook(isbn: Isbn): Book? {
		val bookId = searchBookId(isbn) ?: return null
		val url = "$URL/show/$bookId.json?key=${CONFIG.goodreads}"
		val request = XML.toJSONObject(Util.httpStrRequest(url) ?: return null)
		LOGGER.info("Request: $request")
		return null
	}

	private fun searchBookId(isbn: Isbn): Int?{
		val url = "$URL/isbn_to_id/$isbn?key=${CONFIG.goodreads}"
		return Util.httpStrRequest(url)?.toIntOrNull()
	}
}